source/retryable-writes/tests/etc/templates/handshakeError.yml.template

```diff
@@ -0,0 +1,155 @@
+# Tests in this file are generated from handshakeError.yml.template.
+
+description: "retryable writes handshake failures"
+
+schemaVersion: "1.3"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    auth: true
+
+createEntities:
+  - client:
+      id: &client client
+      useMultipleMongoses: false
+      observeEvents:
+        - connectionCheckOutStartedEvent
+        - commandStartedEvent
+        - commandSucceededEvent
+        - commandFailedEvent
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName retryable-writes-handshake-tests
+  - collection:
+      id: &collection collection
+      database: *database
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  # Because setting a failPoint creates a connection in the connection pool, run
+  # a ping operation that fails immediately after the failPoint operation in
+  # order to discard the connection before running the actual operation to be
+  # tested. The saslContinue command is used to avoid SDAM errors.
+  #
+  # Description of events:
+  # - Failpoint operation.
+  #   - Creates a connection in the connection pool that must be closed.
+  # - Ping operation.
+  #   - Triggers failpoint (first time).
+  #   - Closes the connection made by the fail point operation.
+  # - Test operation.
+  #   - New connection is created.
+  #   - Triggers failpoint (second time).
+  #   - Tests whether operation successfully retries the handshake and succeeds.
+{% for operation in operations %}
+  - description: "{{operation.operation_name}} succeeds after retryable handshake network error"
```

Comment:
Not 100% sure why but in Python this test fails for bulkWrite with a dup key error on `_id:1`:
```
ERROR: test_bulkWrite_succeeds_after_retryable_handshake_network_error (__main__.TestUnifiedHandshakeError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 1647, in test_case
    self.run_scenario(spec)
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 1600, in run_scenario
    self._run_scenario(spec, uri)
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 1624, in _run_scenario
    self.run_operations(spec["operations"])
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 1535, in run_operations
    self.run_entity_operation(op)
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 1276, in run_entity_operation
    result = cmd(**dict(arguments))
  File "/Users/shane/git/mongo-python-driver/pymongo/_csot.py", line 105, in csot_wrapper
    return func(self, *args, **kwargs)
  File "/Users/shane/git/mongo-python-driver/pymongo/collection.py", line 531, in bulk_write
    bulk_api_result = blk.execute(write_concern, session)
  File "/Users/shane/git/mongo-python-driver/pymongo/bulk.py", line 513, in execute
    return self.execute_command(generator, write_concern, session)
  File "/Users/shane/git/mongo-python-driver/pymongo/bulk.py", line 394, in execute_command
    _raise_bulk_write_error(full_result)
  File "/Users/shane/git/mongo-python-driver/pymongo/bulk.py", line 136, in _raise_bulk_write_error
    raise BulkWriteError(full_result)
pymongo.errors.BulkWriteError: batch op errors occurred, full error: {'writeErrors': [{'index': 0, 'code': 11000, 'keyPattern': {'_id': 1}, 'keyValue': {'_id': 1}, 'errmsg': 'E11000 duplicate key error collection: retryable-writes-handshake-tests.coll index: _id_ dup key: { _id: 1 }', 'op': {'_id': 1}}], 'writeConcernErrors': [], 'nInserted': 0, 'nUpserted': 0, 'nMatched': 0, 'nModified': 0, 'nRemoved': 0, 'upserted': []}
```

Perhaps we can change the `_id:1` doc to `x:1` to match the insert/insert_many tests?

Reply:
I overlooked a copy-paste error when copying the reads template to writes. Will address.